lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -429,11 +433,55 @@ function dosomething_user_validate_address_field($form, &$form_state) {
     }
     // We have a full address, save it!
     else {
-      dosomething_user_set_address_values($form, $form_state, $formatted_address);
+      if ($form['#form_id'] === 'dosomething_user_info_form') {
+        dosomething_user_set_address_values_from_info_form($form, $form_state, $formatted_address);
+      }
+      else {
+        dosomething_user_set_address_values_from_edit_form($form, $form_state, $formatted_address);
+      }
     }
```

Comment:
@angaither instead of this if block, I tried to just `return TRUE;` and then write 
```
if (function dosomething_user_validate_address_field($first_name, $last_name, $address, $form, &$form_state)) {
  dosomething_user_set_address_values_from_edit_form($form, $form_state, $formatted_address);
}
```
as the last statement in `function dosomething_user_validate_edit_form_address` (and similar for `function dosomething_user_validate_info_form_address`) but when I did that, it for some reason wouldn't save the address in the address fields - it would successfully validate but would erase these fields and leave them blank on save. 

Tried this way instead to get it to work but not sure your thoughts on it - lmk!